google-ads-examples/src/main/java/com/google/ads/googleads/examples/basicoperations/UpdateExpandedTextAd.java

```diff
@@ -0,0 +1,121 @@
+import com.google.ads.googleads.v3.errors.GoogleAdsException;
+import com.google.ads.googleads.v3.resources.Ad;
+import com.google.ads.googleads.v3.services.AdOperation;
+import com.google.ads.googleads.v3.services.AdServiceClient;
+import com.google.ads.googleads.v3.services.MutateAdResult;
+import com.google.ads.googleads.v3.services.MutateAdsResponse;
+import com.google.ads.googleads.v3.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/** Updates an expanded text ad. To get expanded text ads, run GetExpandedTextAds.cs. */
+public class UpdateExpandedTextAd {
+
+  private static class UpdateExpandedTextAdParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.AD_ID, required = true)
+    private Long adId;
+  }
+
+  public static void main(String[] args) {
+    UpdateExpandedTextAdParams params = new UpdateExpandedTextAdParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new UpdateExpandedTextAd().runExample(googleAdsClient, params.customerId, params.adId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long adId) {
+    // Creates an AdOperation to update an ad.
+    AdOperation.Builder adOperation = AdOperation.newBuilder();
+
+    // Creates an Ad in the update field of the operation.
+    Ad.Builder adBuilder =
```

Comment:
Personally I think this is preferable to creating the objects separately. There's no stopping to figure out what you need next - just create the operation and keep drilling down until you set all the fields you need. Do you feel strongly that this is undesirable?

Reply:
I personally find the other way more intuitive, but don't feel that strongly about it. My bigger concern is consistency across examples. It makes sense to me if we are using this as a way to demonstrate an alternative way of building these examples. However, if I were new to the library and came across this one example that was different than the others, it could be confusing. I'd assume there was a reason that this example was different and not that it was just a style preference.

Reply:
Hmm, I agree consistency is preferable, but I don't think it's so confusing that we should chagne this. I've filed https://github.com/googleads/google-ads-java/issues/284 to capture the consistency going fowards.